crates/languages/src/python.rs

```diff
@@ -196,6 +196,42 @@ pub(super) fn python_task_context() -> ContextProviderWithTasks {
             args: vec![VariableName::File.template_value()],
             ..TaskTemplate::default()
         },
+        TaskTemplate {
+            label: format!("unittest '{}'", VariableName::File.template_value()),
+            command: "python3".to_owned(),
+            args: vec![
+                "-m".to_owned(),
+                "unittest".to_owned(),
+                VariableName::File.template_value(),
+            ],
+            ..TaskTemplate::default()
+        },
+        TaskTemplate {
+            label: "unittest $ZED_SYMBOL TestCase".to_owned(),
+            command: "python3".to_owned(),
+            args: vec![
+                "-m".to_owned(),
+                "unittest".to_owned(),
+                VariableName::RelativeFile.template_value(),
+                "-k".to_owned(),
+                "$ZED_SYMBOL".to_owned(),
+            ],
+            tags: vec!["python-unittest-class".to_owned()],
+            ..TaskTemplate::default()
+        },
+        TaskTemplate {
+            label: "unittest $ZED_SYMBOL function".to_owned(),
+            command: "python3".to_owned(),
+            args: vec![
+                "-m".to_owned(),
+                "unittest".to_owned(),
+                VariableName::RelativeFile.template_value(),
+                "-k".to_owned(),
+                "$ZED_SYMBOL".to_owned(),
+            ],
+            tags: vec!["python-unittest-function".to_owned()],
+            ..TaskTemplate::default()
+        },
```

Comment:
This task for functions looks the same as the that for the `TestCase`. Ideally, `ZED_SYMBOL` should be something like `test_module.TestClass.test_method` which enables precisely targetting the test to run. Currently, if there are two identically named test functions (from different `TestCase` classes) in the same file, both will be run when one of them is pressed, since I'm filtering by file name and function name.

I am not sure if this is a current limitation or that there is some way to construct this string from tree-sitter queries (since I know both the node names of the TestCase class and the name of the function). @osiewicz perhaps you could provide some pointers here?

Reply:
Hey, thanks for submitting this PR; the answer to your woes is: ZED_SYMBOL is probably not what you want to use as extension author. It was one of the first task variables we've exposed and it's quite limited in what it can do (which you pointed out already). 
Yes, it is possible to construct your own environmental variables to expose for tasks. Long story short: any capture not prefixed with `_` will be set in the environment of the task you're spawning under the `ZED_CUSTOM_{CAPTURE_NAME}` name. E.g: a full text of a capture named `food` is stored in `ZED_CUSTOM_food`.
Now, you want to do some processing on the values of the captures as well, right? You want to concatenate them with `.` delimiter. You can do that too. Built in languages (which Python classifies as) can inspect variables from captures and add arbitrary environment variables. Note that they can inspect all variables from captures, including those with `_` prefix.
TL;DR: you can take a look at how Go does it: https://github.com/zed-industries/zed/blob/main/crates/languages/src/go.rs#L453
It implements `ContextProvider` trait which has a `build_context` method. This method takes `task_variables`, which contain the values of captures. It then returns variables which should be appended to that context.

Reply:
I went ahead and created a patch that will allow you to focus on implementing `build_context` instead of wrestling with setting it all up. 
[your_own_context_provider_in_python.patch](https://github.com/user-attachments/files/15521749/your_own_context_provider_in_python.patch)
Give that a try. Pro tip: you can run a task that spawns `env` (it can even be a one-shot task) anywhere in the code to see what environmental variables are available for you at a given point.